src/test/scala/com/microsoft/hyperspace/index/RefreshIndexTests.scala

```diff
@@ -223,6 +221,89 @@ class RefreshIndexTests extends QueryTest with HyperspaceSuite {
     }
   }
 
+  test(
+    "Validate refresh delete action updates appended and deleted files in metadata as" +
+      "expected, when some file gets deleted and some appended to source data.") {
+    withSQLConf(
+      IndexConstants.INDEX_LINEAGE_ENABLED -> "true",
+      IndexConstants.REFRESH_DELETE_ENABLED -> "true") {
+      withIndex(indexConfig.indexName) {
+        SampleData.save(
+          spark,
+          nonPartitionedDataPath,
```

Comment:
Can we also utilize `withTempPathAsString` here? Eventually, I want to get rid of this predefined dir path if not necessary.

Reply:
thanks, @imback82 , added, somehow I forgot it for this test and kept it in the next test. now everything is fixed.